### PR TITLE
[MBL-17332][Teacher] Edit file permissions

### PR DESCRIPTION
**Pull request by hermannakos**
Test plan: See ticket for design. Compare functionality with web.

refs: MBL-17332
affects: Teacher
release note: none

## Checklist

- [x] Tested in dark mode
- [x] Tested in light mode
- [ ] A11y checked

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: apps/teacher/src/main/java/com/instructure/teacher/features/modules/list/ui/ModuleListFragment.kt [hunk 1]
```diff
@@ -19,6 +19,7 @@
 package com.instructure.teacher.features.modules.list.ui
 
 import android.os.Bundle
+import com.instructure.canvasapi2.apis.FileFolderAPI
```
Comment (tamaskozmer) [inline apps/teacher/src/main/java/com/instructure/teacher/features/modules/list/ui/ModuleListFragment.kt:22]: I guess this is unused.

File: apps/teacher/src/main/java/com/instructure/teacher/features/modules/list/ui/ModuleListRecyclerAdapter.kt [hunk 2]
```diff
@@ -17,9 +17,11 @@
 package com.instructure.teacher.features.modules.list.ui
 
 import android.content.Context
+import com.instructure.canvasapi2.models.ModuleContentDetails
 import com.instructure.teacher.adapters.GroupedRecyclerAdapter
 import com.instructure.teacher.adapters.ListItemCallback
 import com.instructure.teacher.features.modules.list.ui.binders.*
+import java.util.Date
```
Comment (tamaskozmer) [inline apps/teacher/src/main/java/com/instructure/teacher/features/modules/list/ui/ModuleListRecyclerAdapter.kt:24]: This might be unused as well.

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Review by tamaskozmer — APPROVED** [review]

File: apps/teacher/src/main/java/com/instructure/teacher/features/modules/list/ModuleListPresenter.kt [hunk 3]
```diff
@@ -132,7 +132,10 @@ object ModuleListPresenter : Presenter<ModuleListModel, ModuleListViewState> {
             indent = item.indent * indentWidth,
             tintColor = courseColor,
             enabled = !loading,
-            isLoading = loading
+            isLoading = loading,
+            type = ModuleItem.Type.valueOf(item.type.orEmpty()),
```
Comment (kristofnemere) [inline apps/teacher/src/main/java/com/instructure/teacher/features/modules/list/ModuleListPresenter.kt:136]: this could crash, could you handle it?

**Review by kristofnemere — CHANGES_REQUESTED** [review]

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Review by kristofnemere — APPROVED** [review]
QA+1

**Pull request merged**